test/spec/util/tile-error-collection.spec.js

```diff
@@ -0,0 +1,196 @@
+
+describe('util/tile-error-collection', function () {
+  var interceptAjaxCall;
+  beforeEach(function () {
+    interceptAjaxCall = null;
+
+    this.error = { type: 'limit', message: 'Some message' };
+    this.tiles = [
+      new Backbone.Model({ url: 'some_url/0/0/0/1.png', tileDomNode: {} }),
+      new Backbone.Model({ url: 'some_url/0/0/0/2.png', tileDomNode: {} }),
+      new Backbone.Model({ url: 'some_url/0/0/0/3.png', tileDomNode: {} })
+    ];
+    this.collection = new TileErrorCollection(this.tiles);
+
+    this.originalAjax = $.ajax;
+    $.ajax = function (params) {
+      interceptAjaxCall && interceptAjaxCall(params);
+      return {
+        complete: function (cb) {
+          cb();
+          return this;
+        },
+        success: function (cb) {
+          cb();
+          return this;
+        },
+        error: function (cb) {
+          cb();
+          return this;
+        }
+      };
+    };
+  });
+
+  describe('._onAdd', function () {
+    var tile = {
+      url: 'some_url/0/0/0/4.png',
+      tileDomNode: {}
+    };
+
+    it('should set the error tile overlay when a new model is added', function () {
+      spyOn(document.body, 'contains').and.returnValue(true);
+      this.collection.add(tile);
+      var model = _.last(this.collection.models);
+      expect(model.get('tileDomNode').src.indexOf('data:image/svg+xml;base64,')).not.toBe(-1);
+    });
+
+    it('should add the model to the qeue if is not there', function () {
+      // Set running to true to prevent ._getTileErrors from being called
+      // since the image is not in the DOM and will be removed from the queue
+      this.collection.running = true;
+
+      this.collection.add(tile);
+      expect(this.collection.queue.length).toEqual(1);
+
+      this.collection.add(tile);
+      expect(this.collection.queue.length).toEqual(1);
+    });
+
+    describe('when running is false', function () {
+      it('should set running to true', function () {
+        this.collection.running = false;
+        this.collection.add(tile);
+        expect(this.collection.running).toBe(true);
+      });
+
+      it('should call ._getTileErrors ', function () {
+        spyOn(this.collection, '_getTileErrors');
+
+        this.collection.running = true;
+        this.collection.add(tile);
+        expect(this.collection._getTileErrors).not.toHaveBeenCalled();
+
+        this.collection.running = false;
+        this.collection.add(tile);
+        expect(this.collection._getTileErrors).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('.getError', function () {
+    it('should return an error if any tile have it', function () {
+      this.collection.models[0].set('error', this.error);
+
+      expect(this.collection.getError()).toEqual(this.error);
+    });
+  });
+
+  describe('.resetErrorTiles', function () {
+    it('should remove the tiles if they\'re not in the DOM', function () {
```

Comment:
This test is a bit tricky to understand without having to know what `_deletedNode` does... Why not add some tiles to the collection and some of them to the DOM and calling `resetErrorTiles`?

Reply:
How would you add them to the DOM here?

Reply:
Couldn't we create `<img>` elements and append them to (or remove them from) `document.body`?

Reply:
I'm not sure, I was having problems with `document.body.containts` because `document.body` was undefined, so I had to mock that call, but I'll give it a try.